astropath/path.py

```diff
@@ -0,0 +1,212 @@
+        if mag is not None:
+            self.candidates['mag'] = mag
+        # Vet
+        assert candidates.vet_candidates(self.candidates), 'Bad candidate input'
+        # 
+        self.init_cand_coords()
+
+    def init_cand_coords(self):
+        """Simple method to generate the coords object
+        from the Table.  
+        
+        In case the candidates are set in some other fashion.
+        """
+        # Add SkyCoord objects
+        self.cand_coords = SkyCoord(ra=self.candidates.ra, 
+                                             dec=self.candidates.dec, 
+                                             unit='deg')
+
+    def init_cand_prior(self, P_O_method:str, P_U=0.):
+        """Ingest the candidate prior approach and P_U
+
+        Args:
+            P_O_method (str): Method for calculation candidate prior
+            P_U (float, optional): Unseen prior. Defaults to 0..
+
+        Raises:
+            IOError: [description]
+        """
+        if self.candidates is None:
+            raise IOError("You must init candidates before their prior")
+        self.cand_prior = dict(P_O_method=P_O_method, 
+                               P_U=P_U)
+        # Vet
+        assert priors.vet_cand_prior(self.cand_prior, self.candidates), 'Bad candidate prior input'
+
+    def init_theta_prior(self, PDF:str, max:float):
+        """Ingest the theta (offset) prior
+
+        Args:
+            PDF (str): Method
+            max (float): Maximum offset allowed
+        """
+        self.theta_prior = dict(PDF=PDF, max=max)
+        # Vet
+        assert priors.vet_theta_prior(self.theta_prior)
+
+    def init_localization(self, ltype:str, **kwargs):
+        """Ingets the localization information
+
+        Args:
+            ltype (str): [description]
+            kwargs: Other parameters definig the localization
+                Depends on the ltype.  See localization.py
+        """
+        self.localiz = dict(type=ltype)
+        self.localiz.update(kwargs)
+        # Vet
+        assert localization.vet_localization(self.localiz), 'Bad candidate prior input'
+
+    def calc_priors(self):
+        """Calculate and normalize the P(O) values for the candidates
+
+        Raises:
+            IOError: [description]
+
+        Returns:
+            np.ndarray: P(O) values.  Also ingested in self.candidates['P_O']
+        """
+        if self.candidates is None or self.cand_prior is None:
+            raise IOError("Init candidates and cand_prior first!")
+
+        # Raw priors
+        self.raw_prior_Oi = priors.raw_prior_Oi(
+            self.cand_prior['P_O_method'], self.candidates['ang_size'], 
+            mag=self.candidates['mag'] if 'mag' in self.candidates.keys() else None)
+
+        # Normalize
+        self.prior_Oi = priors.renorm_priors(self.raw_prior_Oi, 
+                                             self.cand_prior['P_U'])
+
+        # Add to candidate table
+        self.candidates['P_O'] = self.prior_Oi
```

Comment:
Would be useful to add logging/print statements here like: 
"Calculating priors"
"Rernormalizing priors"
"Adding prior values to candidates dataframe"

Reply:
done